src/robotFactory.js

```diff
@@ -1,10 +1,82 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+
+    return this;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+
+    return this;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+
+    return this;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+
+    return this;
+  }
+
+  getInfo() {
+    return 'Robot: ' + this.name
+      + ', Chip version: ' + this.chipVersion
+      + ', Weight: ' + this.weight;
```

Comment:
in this way it'll be clearer
```suggestion
    return `Robot: ${this.name}`
      + `, Chip version: ${this.chipVersion}`
      + `, Weight: ${this.weight}`;
```

Reply:
> return `Robot: ${this.name}`
>       + `, Chip version: ${this.chipVersion}`
>       + `, Weight: ${this.weight}`;

Thanks)